frontend/config/apollo.js

```diff
@@ -18,7 +18,17 @@ const authMiddleware = new ApolloLink((operation, forward) => {
 
 const client = new ApolloClient({
   link: concat(authMiddleware, httpLink),
-  cache: new InMemoryCache()
+  cache: new InMemoryCache({
+    dataIdFromObject: o => {
+      switch (o.__typename) {
+        case 'User':
+          // The primary key of 'User' is 'username'
+          return o.username
```

Comment:
If using username as the primary key proves to be such an issue we can take the time to change it back to the id. We talked offline about this being easier for frontend, but it looks like it's possibly causing issues. @robmcl4 for his feelings about this.